addons/account_peppol/models/account_edi_proxy_user.py

```diff
@@ -104,9 +105,13 @@ def _cron_peppol_get_message_status(self):
         edi_users._peppol_get_message_status()
 
     def _cron_peppol_get_participant_status(self):
-        edi_users = self.search([('company_id.account_peppol_proxy_state', 'in', ['in_verification', 'sender', 'smp_registration'])])
+        waiting_status_update_domain = [('company_id.account_peppol_proxy_state', 'in', ['in_verification', 'sender', 'smp_registration'])]
+        edi_users = self.search(waiting_status_update_domain)
         edi_users._peppol_get_participant_status()
 
+        if self.search(waiting_status_update_domain):
```

Comment:
```suggestion
        if self.search(waiting_status_update_domain, limit=1):
```

Reply:
You can even search_count + limit=1, which avoids the order_by of the search and makes it even more optimal, but it's nitpicking :smirk: